PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = 'hr.contract'
-    _inherit = 'hr.leave.type'
-    _inherit = 'hr.leave'
-    _inherit = 'hr.work.entry'
-    _inherit = 'hr.work.entry.type'

Total Changes: 10

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/hr_work_entry_holidays/models/hr_contract.py
+++ b/addons/hr_work_entry_holidays/models/hr_contract.py
@@ -8,7 +8,7 @@ from odoo.osv.expression import OR
 
 
 class HrContract(models.Model):
-    _inherit = 'hr.contract'
+    _inherit = ['hr.contract']
     _description = 'Employee Contract'
 
     # override to add work_entry_type from leave

--- a/addons/hr_work_entry_holidays/models/hr_leave.py
+++ b/addons/hr_work_entry_holidays/models/hr_leave.py
@@ -12,13 +12,13 @@ from odoo.tools import format_date
 
 
 class HrLeaveType(models.Model):
-    _inherit = 'hr.leave.type'
+    _inherit = ['hr.leave.type']
 
     work_entry_type_id = fields.Many2one('hr.work.entry.type', string='Work Entry Type')
 
 
 class HrLeave(models.Model):
-    _inherit = 'hr.leave'
+    _inherit = ['hr.leave']
 
     def _prepare_resource_leave_vals(self):
         vals = super(HrLeave, self)._prepare_resource_leave_vals()

--- a/addons/hr_work_entry_holidays/models/hr_work_entry.py
+++ b/addons/hr_work_entry_holidays/models/hr_work_entry.py
@@ -7,7 +7,7 @@ from odoo import api, fields, models
 
 
 class HrWorkEntry(models.Model):
-    _inherit = 'hr.work.entry'
+    _inherit = ['hr.work.entry']
 
     leave_id = fields.Many2one('hr.leave', string='Time Off')
     leave_state = fields.Selection(related='leave_id.state')
@@ -101,7 +101,7 @@ class HrWorkEntry(models.Model):
 
 
 class HrWorkEntryType(models.Model):
-    _inherit = 'hr.work.entry.type'
+    _inherit = ['hr.work.entry.type']
     _description = 'HR Work Entry Type'
 
     leave_type_ids = fields.One2many(
